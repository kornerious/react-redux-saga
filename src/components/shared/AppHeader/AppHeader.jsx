import React from 'react';
import { NavLink } from 'react-router-dom';

import './AppHeader.scss';

const AppHeader = () => (
  <div className="app-header">
    <ul className="raw not-mobile">
      <li>
        <NavLink exact to="/">Home</NavLink>
      </li>
      <li>
        <NavLink to="/about">About</NavLink>
      </li>
    </ul>
    <div className="burger-menu mobile">
      <h2>Burger Menu</h2>
    </div>
  </div>
);

export default AppHeader;
