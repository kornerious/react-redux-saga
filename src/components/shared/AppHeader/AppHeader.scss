@import 'styles/_variables.scss';
@import 'styles/_mixins.scss';

.app-header {
  a {
    color: $navLinkColor;
    &:hover {
      color: $navLinkHoverColor;
    }
  }

  a.active {
    color: $navActiveLink;
  }

  .burger-menu {
    color: white;
  }

  @include tablet {
    background-color: red;
  }

  @include desktop {
    
  }
}