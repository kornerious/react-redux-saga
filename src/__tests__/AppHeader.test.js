import React from 'react';
import { shallow } from 'enzyme';

import AppHeader from 'components/shared/AppHeader';

describe('AppHeader', () => {
  it('renders all links', () => {
    const header = shallow(<AppHeader />);
    expect(header.find('li')).toHaveLength(2);
  });

  it('<AppHeader /> should render list of navigation links', () => {
    const header = shallow(<AppHeader />);
    expect(header).toMatchSnapshot();
  });
});
