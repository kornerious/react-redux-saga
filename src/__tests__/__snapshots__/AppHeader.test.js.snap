// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader <AppHeader /> should render list of navigation links 1`] = `
<div
  className="app-header"
>
  <ul
    className="raw not-mobile"
  >
    <li>
      <NavLink
        exact={true}
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li>
      <NavLink
        to="/about"
      >
        About
      </NavLink>
    </li>
  </ul>
  <div
    className="burger-menu mobile"
  >
    <h2>
      Burger Menu
    </h2>
  </div>
</div>
`;
